src/plugins/lvm-dbus.c

```diff
@@ -3623,6 +3623,50 @@ gboolean bd_lvm_vdo_pool_convert (const gchar *vg_name UNUSED, const gchar *pool
     return bd_lvm_is_tech_avail (BD_LVM_TECH_VDO, BD_LVM_TECH_MODE_CREATE | BD_LVM_TECH_MODE_MODIFY, error);
 }
 
+/**
+ * bd_lvm_vdolvpoolname:
+ * @vg_name: name of the VG containing the queried VDO LV
+ * @lv_name: name of the queried VDO LV
+ * @error: (out): place to store error (if any)
+ *
+ * Returns: (transfer full): the name of the pool volume for the @vg_name/@lv_name
+ * VDO LV or %NULL if failed to determine (@error) is set in those cases)
+ *
+ * Tech category: %BD_LVM_TECH_VDO-%BD_LVM_TECH_MODE_QUERY
+ */
+gchar* bd_lvm_vdolvpoolname (const gchar *vg_name, const gchar *lv_name, GError **error) {
+    GVariant *prop = NULL;
+    gchar *segtype = NULL;
+    gchar *pool_obj_path = NULL;
+    gchar *ret = NULL;
+
+    prop = get_lv_property (vg_name, lv_name, "SegType", error);
+    if (!prop)
+        return NULL;
+
+    g_variant_get_child (prop, 0, "s", &segtype);
+    if (g_strcmp0 (segtype, "vdo") != 0) {
+        g_set_error (error, BD_LVM_ERROR, BD_LVM_ERROR_NOEXIST,
```

Comment:
Leaking `prop` and `segtype`. FYI, I think you can use `&s` as a format string and pass `const gchar *` instead. That would save you two lines of code and some CPU cycles!

Reply:
> FYI, I think you can use `&s` as a format string and pass `const gchar *` instead. That would save you two lines of code and some CPU cycles!

Cool, I didn't know this is possible.